### PR TITLE
feat(aria): extend toHaveAccessibleName() to accept an array of expected accessible names

### LINKED ISSUE #32593: [Feature]: Extend toHaveAccessibleName() to accept an array of expected accessible names
### 🚀 Feature Request

It would be great if `toHaveAccessibleName()` would work as `toHaveText()` and accept an array of expected accessible names, not only a single string or regular expression. In other words, `expected: string | RegExp` should be changed to `expected: string | RegExp | (string | RegExp)[]`. 

### Example

With an Excel-like table where the columns are named A, B, ..., and the rows are named 1, 2, ..., asserting the entire table at once would look something like that:

```TypeScript
await expect(page.getByRole("row")).toHaveAccessibleName([
    "Cell A1 Cell B1 Cell C1",
    "Cell A2 Cell B2 Cell C2",
    "Cell A3 Cell B3 Cell C3",
]);
```

### Motivation

To assert the entire table with the current version of `toHaveAccessibleName()`, I have to write more lines that make the test harder to read:

```TypeScript
const rows = [
    "Cell A1 Cell B1 Cell C1",
    "Cell A2 Cell B2 Cell C2",
    "Cell A3 Cell B3 Cell C3",
];
await expect(page.getByRole("row")).toHaveCount(rows.length);
for (let i = 0; i < rows.length; i++) {
    await expect(page.getByRole("row").nth(i)).toHaveAccessibleName(rows[i]);
}
```

Furthermore, I think it would align nicely with existing assertions that have that capability, for example, `toHaveText()`.

### PR DESCRIPTION
**Pull request by aairiian**
extend toHaveAccessibleName() to accept an array of expected accessible names and write the tests for the type

Closes https://github.com/microsoft/playwright/issues/32593.

### CODE REVIEW

**Comment by aairiian** [comment]
> @aairiian please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by aairiian** [comment]
@microsoft-github-policy-service agree

**Review by Skn0tt — COMMENTED** [review]
Thanks for the PR, this looks great! I'll discuss with the team tonight to see if there's anything we need to pay close attention to.

**Comment by Skn0tt** [comment]
> I'll discuss with the team tonight.

We're happy to go forward with this. We have a related feature in the works that's about ARIA snapshots, and that should be strictly better than the array forms of `toMatchText`, `toHaveAccessibleName` et al. So just to warn you - if we end up shipping that in this or one of the following releases, we'll drop this PR in favour of that.

**Comment by aairiian** [comment]
> > I'll discuss with the team tonight.
> 
> We're happy to go forward with this. We have a related feature in the works that's about ARIA snapshots, and that should be strictly better than the array forms of `toMatchText`, `toHaveAccessibleName` et al. So just to let you know - if we end up shipping that in this or one of the following releases, we'll drop this PR in favour of that.

Thank you for getting back to me. I appreciate the time you've taken to review my PR and consider it 😌🙏🏻

File: docs/src/api/class-locatorassertions.md [hunk 1]
```diff
@@ -1236,7 +1236,7 @@ await Expect(locator).toHaveAccessibleNameAsync("Save to disk");
 
 ### param: LocatorAssertions.toHaveAccessibleName.name
 * since: v1.44
-- `name` <[string]|[RegExp]>
+- `name` <[string]|[RegExp]|[Array]<[string]>|[Array]<[RegExp]>>
```
Comment (pavelfeldman) [inline docs/src/api/class-locatorassertions.md:1239]: This should be [Array]<[string]|[RegExp]>

File: packages/playwright/src/matchers/matchers.ts [hunk 2]
```diff
@@ -189,13 +189,20 @@ export function toHaveAccessibleDescription(
...
+  expected: string | RegExp | (string | RegExp)[],
+  options: { timeout?: number, ignoreCase?: boolean, normalizeWhiteSpace?: boolean } = {}
 ) {
-  return toMatchText.call(this, 'toHaveAccessibleName', locator, 'Locator', async (isNot, timeout) => {
-    const expectedText = serializeExpectedTextValues([expected], { ignoreCase: options?.ignoreCase });
-    return await locator._expect('to.have.accessible.name', { expectedText, isNot, timeout });
-  }, expected, options);
+  if (Array.isArray(expected)) {
+    return toEqual.call(this, 'toHaveAccessibleName', locator, 'Locator', async (isNot, timeout) => {
+      const expectedText = serializeExpectedTextValues(expected, { matchSubstring: true, normalizeWhiteSpace: true, ignoreCase: options.ignoreCase });
```
[the remark this answers is not in the document]
Comment (pavelfeldman) [reply packages/playwright/src/matchers/matchers.ts:197]: +1, let's only land the array aspect.

[the remark this answers is not in the document]
Comment (aairiian) [reply packages/playwright/src/matchers/matchers.ts]: I excluded the matchSubstring option from the string serialization

File: docs/src/api/class-locatorassertions.md (hunk 1 above)
Comment (aairiian) [reply docs/src/api/class-locatorassertions.md]: Fixed the type format

**Review by dgozman — APPROVED** [review]
Looks great, thank you for the PR!

**Pull request merged**